packages/server-admin-ui/src/views/ServerConfig/Settings.js

```diff
@@ -32,20 +32,81 @@ class Settings extends Component {
   constructor (props) {
     super(props)
     this.state = {
-      hasData: false
+      hasData: false,
+      restoreFile: null
     }
 
     this.fetchSettings = fetchSettings.bind(this)
     this.handleChange = this.handleChange.bind(this)
+    this.fileChanged = this.fileChanged.bind(this)
     this.handleOptionChange = this.handleOptionChange.bind(this)
     this.handleInterfaceChange = this.handleInterfaceChange.bind(this)
     this.handleSaveSettings = this.handleSaveSettings.bind(this)
+    this.backup = this.backup.bind(this)
+    this.restore = this.restore.bind(this)
   }
 
   componentDidMount () {
     this.fetchSettings()
   }
 
+  fileChanged (event) {
+    this.setState({ ...this.state,
+      restoreFile: event.target.files[0]
+    })
+  }
+
+  backup() {
+    window.location = '/backup'
+    /*
+    fetch(`/backup`, {
+      credentials: 'include',
+      headers: {
+        'Accept': 'application/zip'
+      }
+    })
+      .then(response => {
+        console.log(response)
+      })
+    */
+  }
+
+  restore() {
+    if ( !this.state.restoreFile ) {
+      alert('Please choose a file')
+      return
+    }
+
+    if ( confirm('Are you sure you want restore your settings?') ) {
+      const data = new FormData() 
+      data.append('file', this.state.restoreFile)
+      
+      fetch(`/restore`, {
+        credentials: 'include',
+        method: 'POST',
+        headers: {
+          'Accept': 'application/json'
+        },
+        body: data
+      })
+        .then(response => {
+          if (response.ok) {
+            alert('The backup was restored. Please restart the server.')
```

Comment:
this should trigger a restart, not ask the user to restart

Reply:
I started to do that and then changed my mind, I forget why...